include/irr/builtin/material_compiler/glsl/common.glsl

```diff
@@ -454,36 +456,36 @@ bxdf_eval_t instr_execute_cos_eval_CONDUCTOR(in instr_t instr, in mat2x3 eta, in
 	return DG*fr;
 }
 
-bxdf_eval_t instr_execute_cos_eval_COATING(in instr_t instr, in mat2x4 srcs, in params_t params, in vec3 eta, in vec3 eta2, in irr_glsl_LightSample s, in irr_glsl_AnisotropicMicrofacetCache microfacet, in float DG, in bsdf_data_t data, out vec3 out_diffuse_weight)
+bxdf_eval_t instr_execute_cos_eval_COATING(in instr_t instr, in mat2x4 srcs, in params_t params, in vec3 eta, in vec3 eta2, in irr_glsl_LightSample s, in irr_glsl_AnisotropicMicrofacetCache microfacet, in float DG, in bsdf_data_t data, out float out_weight)
 {
-	float thickness = uintBitsToFloat(data.data[2].z);
-	//vec3 sigmaA = params_getSigmaA(params);
+	//vec3 thickness_sigma = params_getSigmaA(params);
 	
-	vec3 fr = irr_glsl_fresnel_dielectric_frontface_only(eta, microfacet.isotropic.VdotH);
+	vec3 ws_ = irr_glsl_fresnel_dielectric_frontface_only(eta, microfacet.isotropic.VdotH);
+	// TODO include thickness_sigma in diffuse weight computation: exp(sigma_thickness * freePath)
+	// freePath = ( sqrt(refract_compute_NdotT2(NdotL2, rcpOrientedEta2)) + sqrt(refract_compute_NdotT2(NdotV2, rcpOrientedEta2)) )
+	vec3 wd_ = irr_glsl_diffuseFresnelCorrectionFactor(eta, eta2) * (vec3(1.0) - irr_glsl_fresnel_dielectric_frontface_only(eta, currInteraction.isotropic.NdotV)) * (vec3(1.0) - irr_glsl_fresnel_dielectric_frontface_only(eta, s.NdotL));
+	float ws = dot(CIE_XYZ_Luma_Y_coeffs, ws_);
+	float wd = dot(CIE_XYZ_Luma_Y_coeffs, wd_);
+	float w = ws / (ws + wd);
```

Comment:
what is this `w` and `out_Weight` used for?

Reply:
ok its wrong, you cant use the actual weights as the PDF weights, because that works differently

Reply:
out_weight goes to `instr_execute_cos_eval_pdf_COATING` to not compute same thing twice

Reply:
because your generator chooses coat with probability `Fresnel(NdotV)` the PDF should be
`mix(coated_pdf,coat_pdf,Fresnel(NdotV))`, so `out_weight=1.0-Fresnel(NdotV)`

(I'd rather have you swap L/R child nodes around with coat with coated so you don't need to do `1.0-` all the time)